service/controller/drainer_resource_set.go

```diff
@@ -1,54 +1,23 @@
 package controller
 
 import (
-	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
 	"github.com/giantswarm/microerror"
-	"github.com/giantswarm/micrologger"
-	"github.com/giantswarm/operatorkit/controller"
-	"github.com/giantswarm/operatorkit/resource"
-	"github.com/giantswarm/operatorkit/resource/wrapper/metricsresource"
-	"github.com/giantswarm/operatorkit/resource/wrapper/retryresource"
-	"k8s.io/client-go/kubernetes"
+	"github.com/giantswarm/operatorkit/v4/pkg/resource"
+	"github.com/giantswarm/operatorkit/v4/pkg/resource/wrapper/metricsresource"
+	"github.com/giantswarm/operatorkit/v4/pkg/resource/wrapper/retryresource"
 
-	"github.com/giantswarm/kvm-operator/pkg/project"
-	"github.com/giantswarm/kvm-operator/service/controller/key"
 	"github.com/giantswarm/kvm-operator/service/controller/resource/endpoint"
 	"github.com/giantswarm/kvm-operator/service/controller/resource/pod"
 )
 
-type DrainerResourceSetConfig struct {
-	G8sClient versioned.Interface
-	K8sClient kubernetes.Interface
-	Logger    micrologger.Logger
-
-	ProjectName string
-}
-
-func NewDrainerResourceSet(config DrainerResourceSetConfig) (*controller.ResourceSet, error) {
+func newDrainerResources(config DrainerConfig) ([]resource.Interface, error) {
 	var err error
 
-	handlesFunc := func(obj interface{}) bool {
-		p, err := key.ToPod(obj)
-		if err != nil {
-			return false
-		}
-		v, err := key.VersionBundleVersionFromPod(p)
-		if err != nil {
-			return false
-		}
-
-		if v == project.Version() {
-			return true
-		}
-
-		return false
-	}
-
```

Comment:
@giantswarm/sig-operator I would appreciate a check on this removal. We don't have `handlesFunc` anymore so I replaced it with

```diff
			Selector: labels.SelectorFromSet(map[string]string{
-				key.PodWatcherLabel: project.Name(),
+				key.PodWatcherLabel:   project.Name(),
+				label.OperatorVersion: project.Version(),
			}),
in `service/controller/drainer.go`. I believe it's equivalent, but I don't know how to confirm it.